### PR TITLE
Add TUS support to the GCS storage driver

### PR DESCRIPTION
**Pull request by ComfortablyCoding**
## Scope

What's changed:

- Added TUS support for the GCS storage driver

## Potential Risks / Drawbacks

- None

## Review Notes / Questions

- None

---

Fixes SER-320

### CODE REVIEW

File: packages/storage-driver-gcs/src/index.ts [hunk 1]
```diff
@@ -88,6 +109,64 @@ export class DriverGCS implements Driver {
...
+			.createWriteStream({
+				chunkSize: this.preferredChunkSize,
+				uri: context.metadata!['uri'] as string,
+				offset,
+				isPartialUpload: true,
+				resumeCRC32C: context.metadata!['crc32'] as string,
+				metadata: {
+					contentLength: context.size || 0,
+					metadata: {
+						size: context.size || 0,
```
Comment (ComfortablyCoding) [inline packages/storage-driver-gcs/src/index.ts:140]: Any specific metadata we want to be setting on the file? iirc metadata is a required field/param with at least one key/value.

Comment (ComfortablyCoding) [reply packages/storage-driver-gcs/src/index.ts]: Disregard, upon further testing this does not seem to be the case any longer

File: packages/storage-driver-gcs/src/index.ts [hunk 2]
```diff
@@ -88,6 +112,55 @@ export class DriverGCS implements Driver {
...
+		await pipeline(content, stream);
+
+		return bytesUploaded;
+	}
+
+	async finishChunkedUpload(_filepath: string, _context: ChunkedUploadContext) {}
+
+	async deleteChunkedUpload(filepath: string, _context: ChunkedUploadContext) {
+		await this.delete(filepath);
+	}
```
Comment (licitdev) [inline packages/storage-driver-gcs/src/index.ts:163]: I think we should be deleting the chunked upload here 🤔 

```suggestion
	async deleteChunkedUpload(filepath: string, context: ChunkedUploadContext) {
		await this.deleteChunkedUpload(filepath, context);
	}
```

**Review by licitdev — CHANGES_REQUESTED** [review]

File: packages/storage-driver-gcs/src/index.ts (hunk 2 above)
Comment (ComfortablyCoding) [reply packages/storage-driver-gcs/src/index.ts:163]: The proposed change would be a recursive call to this function which is not desired. The current call is correct.

Comment (licitdev) [reply packages/storage-driver-gcs/src/index.ts:163]: You are very right! I had mistaken it for a GCS function 🫣 Thanks!!

**Review by licitdev — APPROVED** [review]

**Pull request merged**